front/src/modules/timesStops/helpers/computeMargins.ts

```diff
@@ -45,14 +57,21 @@ function computeMargins(
     };
   }
 
-  const pathStepId = path[pathStepIndex].id;
-  const theoreticalMargin = getTheoreticalMargin(selectedTrainSchedule, pathStepId);
+  const { theoreticalMargin, isBoundary } = theoreticalMargins[pathStepId];
+
+  // find the next pathStep where constraints are defined
+  let nextIndex = path.length - 1;
 
-  // find the previous pathStep where margin was defined
-  let prevIndex = 0;
-  for (let index = 1; index < pathStepIndex; index += 1) {
-    if (margins.boundaries.includes(path[index].id)) {
-      prevIndex = index;
+  for (let index = pathStepIndex + 1; index < path.length; index += 1) {
+    const curStepId = path[index].id;
+    const curStepSchedule = scheduleByAt[curStepId];
+    if (
+      theoreticalMargins[curStepId]?.isBoundary ||
+      curStepSchedule?.arrival ||
+      curStepSchedule?.stop_for
```

Comment:
checking if `curStepSchedule` is defined is enough
```suggestion
      curStepSchedule
```

Reply:
Thanks!